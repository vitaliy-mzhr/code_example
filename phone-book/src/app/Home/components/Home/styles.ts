import styled from 'styled-components';

const HomeStyles = styled.div`
  height: 100%;
  @media all and (max-width: ${props => props.theme.media.mobileMaxWidthPx}px) {

  }
`;

export default HomeStyles;
