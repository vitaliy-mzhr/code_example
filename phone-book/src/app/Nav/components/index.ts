import NavBar from './NavBar';
import Search from './Search';
import NavButton from './NavButton';

export {
  NavBar,
  Search,
  NavButton,
};