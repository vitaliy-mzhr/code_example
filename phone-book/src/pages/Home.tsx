import * as React from 'react';
import { withRouter, RouteComponentProps, /*Redirect*/ } from 'react-router';
import HomeContainer from '../app/Home/containers/HomeContainer';
import { BaseLayout } from '../app/layouts';
import { Helmet } from 'react-helmet';

interface AppRoute {
  match: any;
}
const HomePage: React.StatelessComponent<RouteComponentProps<AppRoute>> = (props) => (
  <BaseLayout>
    <Helmet title={'All contacts'}/>
    <HomeContainer {...props}/>
  </BaseLayout>
);

export default withRouter(HomePage);
