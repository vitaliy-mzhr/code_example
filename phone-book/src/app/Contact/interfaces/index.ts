import { AppHomeAction } from '../../Home/interfaces';

export interface AppContact {
  id: number;
  name: {
    first: string;
    last: string;
  };
  phone: string[];
  expanded: boolean;
  divider?: string;
}

export interface AppContactDivider {
  divider: string;
}

export interface AppContactAction {
  type: string;
  payload?: any;
}

export interface AppEditContactDispatch {
  fetchContacts: () => AppHomeAction;
  updateContact: (contact: object, id: number) => AppContactAction;
}