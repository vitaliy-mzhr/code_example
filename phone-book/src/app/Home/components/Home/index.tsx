import * as React from 'react';
import HomeStyles from './styles';
import { AppHomeProps } from '../../interfaces';
import { AppSearchDispatch } from '../../../Nav/interfaces';
import { Preloader } from '../../../common';
import { ContactsList } from '../../components';

const Home: React.StatelessComponent<AppHomeProps & AppSearchDispatch> =
  ({contacts, isLoading, toggleContact, deleteContact}) => {
  return (
    <HomeStyles>
      {isLoading
        ? <Preloader/>
        : (
          <ContactsList
            contacts={contacts}
            toggleContact={toggleContact}
            deleteContact={deleteContact}
          />
        )}
    </HomeStyles>
  );
};

export default Home;
