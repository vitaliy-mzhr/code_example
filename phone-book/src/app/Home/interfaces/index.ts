import { AppContact, AppContactAction } from '../../Contact/interfaces';

export interface AppHomeAction {
  type: string;
  payload?: any;
}

export interface AppHomeState {
  isLoading: boolean;
  error: null | string;
  contacts: AppContact[];
}

export interface AppHomeProps {
  isLoading: boolean;
  contacts: AppContact[];
  error?: null | string;
}

export interface AppHomeDispatch {
  fetchContacts: () => AppHomeAction;
  toggleContact: (id: number) => AppContactAction;
  deleteContact: (id: number) => AppContactAction;
}
