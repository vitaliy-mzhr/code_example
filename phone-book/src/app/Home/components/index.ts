import Home from './Home';
import ContactsList from './ContactsList';
import ContactsDivider from './ContactsDivider';

export {
  Home,
  ContactsList,
  ContactsDivider,
};