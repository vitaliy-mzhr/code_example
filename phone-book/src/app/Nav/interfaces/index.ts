import { AppContact, AppContactAction } from '../../Contact/interfaces';
import { AppHomeAction } from '../../Home/interfaces';

export interface AppNavState {
  inputValue: string;
}

export interface AppSearchProps {
  isLoading: boolean;
  error?: null | string;
  contacts: AppContact[];
  searchInput: string;
}

export interface AppNavProps {
  inputValue: string;
}

export interface AppNavAction {
  type: string;
  payload?: string;
}

export interface AppNavDispatch {
  onSearchChange: (input: string) => AppNavAction;
  clearSearchInput: () => AppNavAction;
  fetchContacts: () => AppHomeAction;
}

export interface AppSearchDispatch {
  toggleContact: (id: number) => AppContactAction;
  deleteContact: (id: number) => AppContactAction;
}