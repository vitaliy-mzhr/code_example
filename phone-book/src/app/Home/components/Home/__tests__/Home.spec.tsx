import * as React from 'react';
import { shallow } from 'enzyme';
import Home from '../../Home';
import { AppHomeProps } from '../../../interfaces';
import { AppSearchDispatch } from '../../../../Nav/interfaces';

describe('Home component test', () => {
  it('Home component renders without crashing', () => {
    const mockProps: AppHomeProps & AppSearchDispatch = {
      isLoading: false,
      contacts: [],
      error: null,
      toggleContact: jest.fn(),
      deleteContact: jest.fn(),
    };
    expect(shallow(<Home {...mockProps}/>)).toMatchSnapshot();
  });
  it('Home component renders when loading', () => {
    const mockProps: AppHomeProps & AppSearchDispatch = {
      isLoading: true,
      contacts: [],
      error: null,
      toggleContact: jest.fn(),
      deleteContact: jest.fn(),
    };
    expect(shallow(<Home {...mockProps}/>)).toMatchSnapshot();
  });
});
