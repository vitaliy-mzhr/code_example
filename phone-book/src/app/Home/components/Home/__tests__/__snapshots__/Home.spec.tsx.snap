// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component test Home component renders when loading 1`] = `
<styled.div>
  <Preloader />
</styled.div>
`;

exports[`Home component test Home component renders without crashing 1`] = `
<styled.div>
  <ContactsList
    contacts={Array []}
    deleteContact={[MockFunction]}
    toggleContact={[MockFunction]}
  />
</styled.div>
`;
